In [87]:
import pandas as pd
import os
import re
import shutil
from src.rename_move_files import *

ModuleNotFoundError: No module named 'src'

In [ ]:
"""
# Obtain pathnames of each image
path = "/home/flori/ironhack/projects/final-project"

fullpath_MSI = f"{path}/MSIMUT_JPEG/*.jpg"
fullpath_MSS = f"{path}/MSS_JPEG/*.jpg"
print(fullpath_MSI)
print(fullpath_MSS)

MSIfiles = glob.glob(fullpath_MSI)
MSSfiles = glob.glob(fullpath_MSS)
"""

In [ ]:
"""Two ways to rename files
import shutil
shutil.move('a.txt', 'b.kml')


old_file = os.path.join("/home/flori/ironhack/projects/final-project/MSIMUT_JPEG", "blk-AAADECQEWVSD-TCGA-CM-4746-01Z-00-DX1.jpg")
new_file = os.path.join("../input/biopsy_images", "MSI/1.jpg")
os.rename(old_file, new_file)
"""

In [82]:
"""
No puedo cambiar y mover el archivo a la vez
#rename files and move them
old_file = os.path.join("/home/flori/ironhack/projects/final-project/MSIMUT_JPEG", "blk-AAADECQEWVSD-TCGA-CM-4746-01Z-00-DX1.jpg")
new_file = os.path.join("../input/biopsy_images", "MSI/1.jpg")
os.rename(old_file, new_file)
"""

'\nNo puedo cambiar y mover el archivo a la vez\n#rename files and move them\nold_file = os.path.join("/home/flori/ironhack/projects/final-project/MSIMUT_JPEG", "blk-AAADECQEWVSD-TCGA-CM-4746-01Z-00-DX1.jpg")\nnew_file = os.path.join("../input/biopsy_images", "MSI/1.jpg")\nos.rename(old_file, new_file)\n'

In [3]:
# First rename, then copy with shutil
# Obtain pathnames of each image
path = "/home/flori/ironhack/projects/final-project"

fullpath_MSI = f"{path}/MSIMUT_JPEG/*.jpg"
fullpath_MSS = f"{path}/MSS_JPEG/*.jpg"
print(fullpath_MSI)
print(fullpath_MSS)

MSIfiles = glob.glob(fullpath_MSI)
MSSfiles = glob.glob(fullpath_MSS)

/home/flori/ironhack/projects/final-project/MSIMUT_JPEG/*.jpg
/home/flori/ironhack/projects/final-project/MSS_JPEG/*.jpg


In [4]:
print(MSIfiles[0])

/home/flori/ironhack/projects/final-project/MSIMUT_JPEG/blk-DIIMCHYEWATG-TCGA-AA-3715-01Z-00-DX1.jpg


In [5]:
display(len(MSIfiles), len(MSSfiles))

75039

117273

In [24]:
"""
def checkNameFile(nameFiles, pattern, nameFolder):
    # check if all files contain the sequence "blk-"
    name_list = []
    for name in nameFiles:
        name_list.append(re.search(pattern, name).group())
    return print(f"The number of files in {nameFolder} whose name contains the sequence {pattern} is {len(name_list)}")
    
"""

In [25]:
"""
checkNameFile(MSIfiles, r'MSIMUT_JPEG/blk-', "MSIfiles")
checkNameFile(MSSfiles, r'MSS_JPEG/blk-', "MSSfiles")
"""

The number of files in MSIfiles whose name contains the sequence MSIMUT_JPEG/blk- is 75039
The number of files in MSSfiles whose name contains the sequence MSS_JPEG/blk- is 117273


In [33]:
"""
# All files contain the pattern "*_JPEG/blk-". Use this sequence to rename files
# 
def renameFiles(nameFiles, name):
    for i in range(len(nameFiles)):
        old_name = os.path.join(nameFiles[i].split("/blk-")[0], nameFiles[i].split("_JPEG/")[1])
        print(old_name)
        new_name = os.path.join(nameFiles[i].split("/blk-")[0], f'{name}_{str(i)}.jpg')
        print(new_name)
        os.rename(old_name, new_name)
    return print("File names changed")
"""

In [55]:
#Creo que con listdir es más sencillo y limpio, voy a intentarlo. Voy a probarlo sobre una carpeta de prueba con solo 3 archivos
path = "/home/flori/ironhack/projects/final-project/prueba_cambio_nombre"
pruebafiles = os.listdir(path)
print(pruebafiles)

['blk-AAADECQEWVSD-TCGA-CM-4746-01Z-00-DX1.jpg', 'blk-AAAFIYHTSVIE-TCGA-G4-6309-01Z-00-DX1.jpg', 'blk-AAADTPTGAKQV-TCGA-A6-3809-01Z-00-DX1.jpg']


In [56]:
def renameFiles(path, files, nameFile):
    count = 0
    for i in range(len(files)):
        if re.search(fr"^{nameFile}", files[i])==None:
            old_name = os.path.join(path, files[i])
            new_name = os.path.join(path, f'{nameFile}_{str(i)}.jpg')
            os.rename(old_name, new_name)
            count += 1
    return f"Number of files that have been renamed: {count}"

In [57]:
renameFiles(path, pruebafiles, "MSI")

'Number of files that have been renamed: 3'

In [58]:
pruebafiles = os.listdir(path)
print(pruebafiles)

['MSI_1.jpg', 'MSI_2.jpg', 'MSI_0.jpg']


In [ ]:
#Funciona bien con esta carpeta de 3 archivos. Voy probar con las carpetas que contienen las imagenes

In [60]:
# cambio nombre a todos los archivos de MSI y MSS
path_comun = "/home/flori/ironhack/projects/final-project"
pathMSI = f"{path_comun}/MSIMUT_JPEG"
pathMSS = f"{path_comun}/MSS_JPEG"
MSIfiles = os.listdir(pathMSI)
MSSfiles = os.listdir(pathMSS)
print(len(MSIfiles),MSIfiles[0])
print(len(MSSfiles),MSSfiles[0])

75039 blk-DIIMCHYEWATG-TCGA-AA-3715-01Z-00-DX1.jpg
117273 blk-WAIVQLAAWVDN-TCGA-CA-6718-01Z-00-DX1.jpg


In [61]:
renameFiles(pathMSI, MSIfiles, "MSI")
renameFiles(pathMSS, MSSfiles, "MSS")

'Number of files that have been renamed: 117273'

**Una vez cambiado el nombre de todos los archivos (los 75039 de MSI y los 117273 de MSS) voy a cambiarlos de carpeta**

In [2]:
# create a new folder which will contain all images. Each image will be renamed such as MSI/1 or MSS/2
new_path = "../input/biopsy_images"
if not os.path.isdir(new_path):
    os.makedirs(new_path)

In [68]:
# Una vez cambiados los nombres de los archivos vamos a moverlos a la nueva carpeta creada 
# shutil.move mueve el archivo desde una ubicación a otra
#Voy a probar de nuevo con la carpeta de prueba

def moveFiles(old_path,files_in_path, new_path):
    count = 0
    for file in files_in_path:
        shutil.move(os.path.join(old_path, file), os.path.join(new_path, file))
        count += 1
    return print(f"Number of files that have been moved: {count}")

In [66]:
path = "/home/flori/ironhack/projects/final-project/prueba_cambio_carpeta"
pruebafiles = os.listdir(path)
print(pruebafiles)

['MSI_1.jpg', 'MSI_2.jpg', 'MSI_0.jpg']


In [71]:
moveFiles(path, pruebafiles, new_path)

Number of files that have been moved: 3


In [72]:
pruebafiles = os.listdir(path)
print(pruebafiles)

[]


In [73]:
pruebafiles = os.listdir(new_path)
print(pruebafiles)

['MSI_1.jpg', 'MSI_2.jpg', 'MSI_0.jpg']


In [ ]:
#elimino estos 3 archivos para que no aparezcan en la carpeta de las imagenes

In [80]:
for file in pruebafiles:
    os.remove(f"{new_path}/{file}")

FileNotFoundError: [Errno 2] No such file or directory: '../input/biopsy_images/MSI_1.jpg'

In [81]:
pruebafiles = os.listdir(new_path)
print(pruebafiles)

[]


**Ahora que ya he borrado todo lo de prueba en la carpeta biopsy_files, voy a mover los archivos finales ya renombrados a esta carpeta**

In [ ]:
"""
pics = dict()
for image in MSIfiles:
    label = image.split("/")[-2]
    pic = image.split("/")[-2]+"/"+foto.split("/")[-1]
    if label in pics:
        pics[label].append(pic)
    else:
        pics[label] = [pic]
"""